logic/path/src/selectors/mod.rs

```diff
@@ -14,7 +14,7 @@ where
     W: Default + Add<W, Output = W>,
 {
     /// Edge weighting function.
-    fn calculate_weight(channel: &ChannelEntry) -> W;
+    fn calculate_weight(channel: &ChannelEntry) -> Result<W>;
```

Comment:
Calculating the weight should not fail. If it does, the trait bound on `W` requires `Default`, so the `calculate_weight` function should use that.